components/brave_wallet/common/solana_utils_unittest.cc

```diff
@@ -189,4 +189,26 @@ TEST(SolanaUtilsUnitTest, Uint8ArrayDecode) {
   }
 }
 
+TEST(SolanaUtilsUnitTest, CompactArrayDecode) {
```

Comment:
Is it possible for the byte_index to be negative? 
Could the byte array have a null byte in between it as well?

I'm trying to think what other cases we might want to test for here to make sure we're decoding safely. Much of the safety of the application logic rests on these lower level decode functions being safe since we're parsing untrusted inputs so I'd like to make sure we're not missing any edge cases here.

@iefremov would be good to add this function to the fuzzing list if we've already got that functionality in.

Reply:
It won't be negative, the parameter is size_t (unsigned).

Reply:
I'm not sure what you meant about "have a null byte in between", our byte array is `std::vector<uint8_t>`, each element's possible value is 0-UINT8_MAX.

Reply:
It's impossible to have negative index due to its type. Even if you assign -1 to size_t, compiler will do 2's compliment for it and that will be 8 bytes full of 1 bit which is 18446744073709551615 (max of size_t)

Reply:
Ok, so from the sounds of it the type system should solve for the edge cases I could come up with here. I can't come up with any others that might cause it to fail so going to resolve this one too.